WEB-INF/classes/gov/noaa/pfel/erddap/handlers/AddAttributesHandler.java

```diff
@@ -0,0 +1,79 @@
+package gov.noaa.pfel.erddap.handlers;
+
+import com.cohort.array.CharArray;
+import com.cohort.array.PAType;
+import com.cohort.array.PrimitiveArray;
+import com.cohort.array.StringArray;
+import com.cohort.util.String2;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+
+public class AddAttributesHandler extends State {
+    private StringBuilder content = new StringBuilder();
+    private com.cohort.array.Attributes tAttributes = new com.cohort.array.Attributes();
+    private String tName = null, tType =  null;
+    private State completeState;
+
+    public AddAttributesHandler(SaxHandler saxHandler, com.cohort.array.Attributes tAddGlobalAttributes, State completeState) {
+        super(saxHandler);
+        tAddGlobalAttributes = this.tAttributes;
```

Comment:
Is this supposed to be `this.tAttributes = tAddGlobalAttributes;`

As this is written I believe this line has no effect.

Reply:
So `tAddGlobalAttributes` is actually null and `this.tAttributes` is pointing to the object which has the parsing results needed which is initialised a few lines above this.

As for why is it named `tAddGlobalAttributes` and not just `tAttributes`.
If we take a look at a class like `EDDGridFromDap` you can see a line like,
```
else if (localTags.equals("<addAttributes>"))
        tGlobalAttributes = getAttributesFromXml(xmlReader);
```

But maybe changing the argument name to tAttributes might be better here?
That would make the line look like `tAttributes = this.tAttributes`